charts/pulsar/templates/broker/broker-statefulset.yaml

```diff
@@ -62,7 +62,7 @@ spec:
 {{- with .Values.broker.securityContext }}
 {{ toYaml . | indent 8 }}
 {{- end }}
-    {{- if or .Values.broker.serviceAccount.use .Values.components.functions }}
+    {{- if or .Values.broker.serviceAccount.use (and .Values.components.functions  (not .Values.functions.useDedicateRunner)) }}
```

Comment:
The change looks to be equivalent to the original code. I'm just curious if there are other use cases that also requires a dedicated service account for brokers, like offloading data to the object storage with some workload identities.

Reply:
yeah, offloading could be the other use case, but for now just trying to keep the same behavior as before.